precompiles/types/coin_test.go

```diff
@@ -20,11 +20,33 @@ func Test_createCoinSet(t *testing.T) {
 	tokenDenom := ZRC20ToCosmosDenom(tokenAddr)
 	amount := big.NewInt(100)
 
-	coinSet, err := CreateCoinSet(tokenAddr, amount)
+	coinSet, err := CreateZRC20CoinSet(tokenAddr, amount)
 	require.NoError(t, err, "createCoinSet should not return an error")
 	require.NotNil(t, coinSet, "coinSet should not be nil")
 
 	coin := coinSet[0]
 	require.Equal(t, tokenDenom, coin.Denom, "coin denom should be %s, got %s", tokenDenom, coin.Denom)
 	require.Equal(t, amount, coin.Amount.BigInt(), "coin amount should be %s, got %s", amount, coin.Amount.BigInt())
 }
+
+func Test_CoinIsZRC20(t *testing.T) {
+	test := []struct {
+		denom    string
+		expected bool
+	}{
+		{"", false}, // Empty string.
+		{"zrc20/", false}, // Missing address.
```

Comment:
If the right part should be address, it seems we're missing a test case `zrc20/<invalid address>`
```
zrc20/foo
```

Reply:
Added in 2b130fdc0f29a4d958d599d633c42f3987a6e7b5